### ISSUE TITLE
Wikipedia plugin crash depending on article media

### ISSUE THREAD

**Issue by sunrosa** [koreader/koreader#12429]
* KOReader version: 2024.07
* Device: Xubuntu

#### Issue
Wikipedia plugin causes crashes on viewing articles in "open full" mode without even downloading them (those that I think have gifs as their first image).
I'm on VirtualBox emulating Xubuntu 24.04 using VMSVGA with 3D acceleration.

#### Steps to reproduce
Try viewing the "Reforms of Russian orthography" article (guaranteed crash), or anything relating to chemistry and quantum physics (some articles crash)

##### `crash.log` (if applicable)
Doesn't exist. I have verbose logging enabled. Don't know why. I searched my entire (and only) drive recursively for it.

**Comment by sunrosa** [comment]
If I'm missing certain dependencies it isn't clear which ones. If I could find the log it would help.

**Comment by benoit-pierre** [comment]
Please provide some traces.

**Comment by benoit-pierre** [comment]
Start in a terminal.

**Comment by sunrosa** [comment]
Sorry i forgot you could do that lol. This is the tail.

```
08/30/24-03:33:15 DEBUG triggering refresh {
  mode = "flashui",
  region = "1906x1020+7+5"
} --[[table: 0x7871eb4c0978]] 
08/30/24-03:33:15 DEBUG refresh on logical rectangle 1906x1020+7+5 
08/30/24-03:33:15 DEBUG not painting 1 covered widget(s) 
08/30/24-03:33:15 DEBUG UIManager:_standbyTransition -> AllowStandby 
08/30/24-03:33:15 DEBUG UIManager:preventStandby, counter increased to 1 
08/30/24-03:33:15 DEBUG fetching https://upload.wikimedia.org/wikipedia/commons/7/7a/Russian_alphabet_%28marks_by_Peter_I%29%2C_page_5.gif 
08/30/24-03:33:15 DEBUG show widget: table: 0x7871eb4c3aa8 
08/30/24-03:33:15 DEBUG setDirty nil from widget table: 0x7871eb4c3aa8 w/ NO region; dithering: nil 
08/30/24-03:33:15 DEBUG not painting 1 covered widget(s) 
08/30/24-03:33:15 DEBUG not painting 1 covered widget(s) 
08/30/24-03:33:15 DEBUG UIManager:_standbyTransition -> PreventStandby 
08/30/24-03:33:15 DEBUG subprocess_done: false  stuff_to_read: false 
08/30/24-03:33:15 DEBUG process not yet done, will check again soon 
08/30/24-03:33:15 DEBUG not painting 1 covered widget(s) 
08/30/24-03:33:15 DEBUG not painting 1 covered widget(s) 
08/30/24-03:33:15 DEBUG subprocess_done: true  stuff_to_read: true 
08/30/24-03:33:15 DEBUG close widget: table: 0x7871eb4c3aa8 
08/30/24-03:33:15 DEBUG Lower widget table: 0x7871ee4b3ee0 covers the full screen 
08/30/24-03:33:15 DEBUG not refreshing 1 covered widget(s) 
08/30/24-03:33:15 DEBUG  fetched 7844 
08/30/24-03:33:15 DEBUG GIF file provided, renderImageData: using GifLib 
ffi.load: libs/libgif.so.7
/usr/bin/koreader: line 25: 12095 Segmentation fault      (core dumped) ./reader.lua "${ARGS}"
```

**Comment by benoit-pierre** [comment]
> libgif7 is already installed.

KOReader should be using it's own copy anyway. You're using the AppImage, right?

Can you post the output when running with `LD_DEBUG=files` in the environment.

**Comment by Frenzie** [comment]
Is that Xubuntu 24.04?

**Comment by sunrosa** [comment]
Yes it is 24.04

Here is the tail with LD_DEBUG=files
```
08/30/24-19:10:58 DEBUG subprocess_done: true  stuff_to_read: true 
08/30/24-19:10:58 DEBUG close widget: table: 0x7fb7152e4668 
08/30/24-19:10:58 DEBUG Lower widget table: 0x7fb71a710150 covers the full screen 
08/30/24-19:10:58 DEBUG  fetched 7844 
08/30/24-19:10:58 DEBUG GIF file provided, renderImageData: using GifLib 
     16416:	/usr/lib/koreader/luajit: error: symbol lookup error: undefined symbol: luaJIT_BC_ffi/pic (fatal)
     16416:	/usr/lib/koreader/luajit: error: symbol lookup error: undefined symbol: luaJIT_BC_ffi/giflib_h (fatal)
ffi.load: libs/libgif.so.7
     16416:	
     16416:	file=libs/libgif.so.7 [0];  dynamically loaded by /usr/lib/koreader/luajit [0]
     16416:	file=libs/libgif.so.7 [0];  generating link map
     16416:	  dynamic: 0x00007fb71f487df8  base: 0x00007fb71f47e000   size: 0x000000000000a1e8
     16416:	    entry: 0x00007fb71f480740  phdr: 0x00007fb71f47e040  phnum:                 10
     16416:	
     16416:	
     16416:	calling init: libs/libgif.so.7
     16416:	
     16416:	opening file=libs/libgif.so.7 [0]; direct_opencount=1
     16416:	
/usr/bin/koreader: line 25: 16416 Segmentation fault      (core dumped) ./reader.lua "${ARGS}"
     16413:	
     16413:	calling fini:  [0]
     16413:	
     16413:	
     16413:	calling fini: /lib/x86_64-linux-gnu/libtinfo.so.6 [0]
     16413:	
     16413:	
     16413:	calling fini: /lib/x86_64-linux-gnu/libc.so.6 [0]
     16413:	
     16413:	
     16413:	calling fini: /lib64/ld-linux-x86-64.so.2 [0]
     16413:	
```

**Comment by benoit-pierre** [comment]
Sorry, in this case I need the full log. Also, can you actually directly open GIF files without issues?

**Comment by sunrosa** [comment]
I am able to open gif files alone but not the wikipedia articles.

This is what happens when opening a gif file. The gif does not move which I assume is intended behavior.
[koreader-success.txt](https://github.com/user-attachments/files/16822597/koreader-success.txt)

This is the crash from opening the wikipedia page.
[koreader-crash.txt](https://github.com/user-attachments/files/16822600/koreader-crash.txt)

**Comment by benoit-pierre** [comment]
Nothing look wrong with respect to loaded libraries. Are you willing to build from source so we can get a backtrace?

**Comment by sunrosa** [comment]
I built from source and the bug disappeared.......... I'll try to reproduce it

**Comment by sunrosa** [comment]
Fixed in 2024.11 AppImage on xubuntu (original setup was apt installed 2024.07 on xubuntu)
